static/docs/command-reference/run.md

```diff
@@ -52,25 +52,23 @@ captures data and <abbr>caches</abbr> relevant <abbr>data artifacts</abbr> along
 the way. See [this example](/doc/get-started/example-pipeline) to learn more and
 try creating a pipeline.
 
-### Well-behaved commands
+### Unsupported command behavior
 
-DVC is simple to use, you only need to wrap your commands with `dvc run`, and
-define your dependencies and outputs.
+We don't want to tell you how to write your code! However, please be aware that
+in order to prevent unexpected results when DVC executes or reproduces your
+commands, they should ideally follow these rules:
 
-However, to prevent unexpected behaviors, ideally, your commands should follow
-some principles:
+- Read/write exclusively from/to the specified dependencies (input files) and
+  <abbr>output files</abbr>.
+- Completely rewrite outputs (i.e. do not append or edit).<br/> ⚠️ DVC deletes
+  output files before executing commands, including at `dvc repro`.
+- Stop reading and writing files when the `command` exits.
 
-- Read exclusively from specified dependencies.
-- Write exclusively to specified outputs.
-- Completely rewrite the outputs (i.e. do not append or edit).
-- Stop reading and writing when the command exits.
-
-To guarantee reproducibilty, your command should be
-[deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm) (i.e. it
-must produce the same results given the same inputs/dependencies).
-
-Have in mind what brings entropy to your command (e.g. random generators, time,
-hardware, etc.) and try to minize it (e.g. fix seeds).
+At the very least, if you want to guarantee exact reproducibility, your code
+should be [deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm)
+(i.e. always produce the same output for a given input). So avoid code that
+bring [entropy](https://en.wikipedia.org/wiki/Software_entropy) into your data
+pipeline (e.g. random numbers, time functions, hardware dependency, etc.)
```

Comment:
> `At the very least` -> `Ideally`. Completely deterministic code is extremely hard to achieve in ML

@shcheklein updated.

> `to guarantee reproducibility` - there are different levels of reproducibility - thus I would not tie it directly with deterministic algorithms

That's why I wrote "exact reproducibility". But can you look at the new text and suggest a change? I'm not sure how to improve it. This link between reproducibility and determinism basically comes from Alex and Ramon's version previously merged.

Reply:
I would still omit "exact" ... it sounds like that you should read and care this only if you need the "exact" one. But usually you just have an approximate reproducibility and it can be enough.

Reply:
How about "Keep in mind that if the pipeline's reproducibility goals include consistent
output data, ..." ?

Reply:
Updated.